src/app/components/category-products/category-products.component.ts

```diff
@@ -50,15 +55,37 @@ export class CategoryProductsComponent {
   readonly ratingByStar: FormGroup = new FormGroup({
     rating: new FormControl(),
   });
+  readonly filterByStore: FormGroup = new FormGroup({});
 
-  
   readonly filterByPriceValueChanges$: Observable<any> =
     this.filterProductsByPrice.valueChanges.pipe(
       startWith({ priceFrom: 0, priceTo: 9999 }),
       shareReplay(1)
     );
   readonly filterByStarRatingValueChanges$: Observable<any> =
-    this.ratingByStar.valueChanges.pipe(startWith(0),shareReplay(1));
+    this.ratingByStar.valueChanges.pipe(startWith(0), shareReplay(1));
+
+  readonly stores$: Observable<StoreModel[]> = this._storesService
+    .getAllStores()
+    .pipe(
+      tap((data) => this.createFormControl(data)),
+      shareReplay(1)
+    );
+  createFormControl(stores: StoreModel[]) {
+    stores.forEach((store) =>
+      this.filterByStore.addControl(store.id, new FormControl(false))
+    );
+  }
+
+  readonly filterByStoreValueChanges$: Observable<any> =
+    this.filterByStore.valueChanges.pipe(
+      map((store) => {
+        return Object.keys(store).filter((id) =>
+          store[id] == true ? store[id] : ''
```

Comment:
It should just be "store[id] == true" because the filter method returns the item when the condition is true and skips the item when the condition is false. In your case it looks like when store[id] is true it returns store[id], but when it is not true it returns ' '. This may work, but the empty string may be true in some cases.